Double free of OCSP extensions when a response is parsed twice (GnuPG T8475).
https://github.com/gpg/libksba/commit/c5dc198f571c26b70a6443ac9a2809d4eea16df2

From c5dc198f571c26b70a6443ac9a2809d4eea16df2 Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Tue, 6 Oct 2026 11:11:35 +0900
Subject: [PATCH] Fix cleanup with release_ocsp_extensions.

* src/ocsp.c (ksba_ocsp_parse_response): Clear field
after calling release_ocsp_extensions.

--

GnuPG-bug-id: 8475
Reported-by: Bar Hofesh <bar.hofesh@brightsec.com>
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
---
 src/ocsp.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/ocsp.c b/src/ocsp.c
index ffb3bd4..7b94973 100644
--- a/src/ocsp.c
+++ b/src/ocsp.c
@@ -1496,8 +1496,9 @@ ksba_ocsp_parse_response (ksba_ocsp_t ocsp,
      request. This is useful in case of a TryLater response status. */
   ocsp->response_status = KSBA_OCSP_RSPSTATUS_NONE;
   release_ocsp_certlist (ocsp->received_certs);
-  release_ocsp_extensions (ocsp->response_extensions);
   ocsp->received_certs = NULL;
+  release_ocsp_extensions (ocsp->response_extensions);
+  ocsp->response_extensions = NULL;
   ocsp->hash_length = 0;
   ocsp->bad_nonce = 0;
   xfree (ocsp->responder_id.name);
@@ -1512,6 +1513,7 @@ ksba_ocsp_parse_response (ksba_ocsp_t ocsp,
       *ri->revocation_time = 0;
       ri->revocation_reason = 0;
       release_ocsp_extensions (ri->single_extensions);
+      ri->single_extensions = NULL;
     }
 
   /* Run the actual parser.  */
